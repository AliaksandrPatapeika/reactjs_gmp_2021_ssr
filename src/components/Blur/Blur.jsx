import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const BlurWrapper = styled.div`
  height: 100%;
  backdrop-filter: blur(5px);
  padding: 22px 62px 0;
`;

const Blur = ({children}) => (
  <BlurWrapper>{children}</BlurWrapper>
);

Blur.propTypes = {
  children: PropTypes.node.isRequired
};

export default Blur;
