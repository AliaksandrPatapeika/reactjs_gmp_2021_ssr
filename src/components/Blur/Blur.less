.blur {
	height: 100%;
	backdrop-filter: blur(5px);
	padding: 22px 62px 0;
}
