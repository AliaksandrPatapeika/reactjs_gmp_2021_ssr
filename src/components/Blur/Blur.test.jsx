import React from 'react';
import renderer from 'react-test-renderer';

import Blur from './Blur';

describe('Blur', () => {
  test('renders correctly', () => {
    const blur = renderer
      .create(<Blur>Test</Blur>)
      .toJSON();

    expect(blur).toMatchSnapshot();
  });
});
