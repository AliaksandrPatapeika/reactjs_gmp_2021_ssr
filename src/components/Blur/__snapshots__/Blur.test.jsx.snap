// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blur renders correctly 1`] = `
<div
  className="Blur__BlurWrapper-b9kkbo-0 jwekOJ"
>
  Test
</div>
`;
